src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -135,8 +134,10 @@ void ExecuteWithAbi (string supportedAbis, string apkInputPath, string apkOutput
 				AddNativeLibrariesFromAssemblies (apk, supportedAbis);
 				apk.Flush();
 
-				foreach (ITaskItem typemap in TypeMappings) {
-					apk.Archive.AddFile (typemap.ItemSpec, Path.GetFileName(typemap.ItemSpec), compressionMethod: CompressionMethod.Store);
+				if (TypeMappings != null) {
+					foreach (ITaskItem typemap in TypeMappings) {
+						apk.Archive.AddFile (typemap.ItemSpec, Path.GetFileName(typemap.ItemSpec), compressionMethod: CompressionMethod.Store);
```

Comment:
This will require changes to `embedded-assemblies.cc` in order for this to work.  Currently, it only looks for `typemap.jm` and `typemap.mj` files: https://github.com/xamarin/xamarin-android/blob/f1359d99bddf59802e57479b5c4187f75426e3c9/src/monodroid/jni/embedded-assemblies.cc#L407-L414

Reply:
The code here hasn't changed (and already works). Its just been indented since we made the `TypeMappings` optional rather than `[Required]` as a result we need to check for a `null` value.

So the type maps files we put in the apk will still be called `typemap.jm` and `typemap.mj`

Reply:
I've changed this over to use `ends_with`